src/components/Issue/CreateIssue.tsx

```diff
@@ -0,0 +1,269 @@
+            />
+            <div className="input-title">Description</div>
+            <TextField
+              fullWidth
+              required
+              variant="outlined"
+              margin="normal"
+              multiline
+              minRows={6}
+              name="name"
+              placeholder="Enter description..."
+              helperText={
+                this.state.isSubmitting && this.state.descriptionError
+                  ? this.state.descriptionError
+                  : `Max 255 characters, ${this.state.charactersLeft} left`
+              }
+              error={
+                this.state.isSubmitting && this.state.descriptionError !== null
+              }
+              onChange={this.handleDescriptionChange}
+            />
+            <div className="select-label">Assignee</div>
+            <FormControl>
+              <Select
+                value={this.state.assignee}
+                displayEmpty
+                variant="outlined"
+                onChange={this.handleAssigneeChange}
+                error={
+                  this.state.isSubmitting && this.state.assigneeError !== null
+                }
+              >
+                <MenuItem value="">
+                  <em>--select assignee--</em>
+                </MenuItem>
+                {this.props.users.map((user) => (
+                  <MenuItem value={user.name}>{user.name}</MenuItem>
+                ))}
+              </Select>
+              <FormHelperText
+                error={
+                  this.state.isSubmitting && this.state.assigneeError !== null
+                }
+              >
+                {this.state.isSubmitting && this.state.assigneeError}
+              </FormHelperText>
+            </FormControl>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={this.handleClose} color="primary">
+              Cancel
+            </Button>
+            <Button onClick={this.handleSubmit} color="primary">
+              Save
+            </Button>
+          </DialogActions>
+        </Dialog>
+      </div>
+    );
+  }
+
+  handleSubmit = (): void => {
+    const { titleError, descriptionError, assigneeError } = this.state;
+
+    this.setState(
+      {
+        isSubmitting: true,
+      },
+      () => this.handleValidation()
+    );
+
+    if (titleError || descriptionError || assigneeError) {
+      return;
+    }
+
+    const { title, description, assignee } = this.state;
+
+    this.props.dispatch(createIssue({ title, description, assignee }));
+
+    this.handleClose();
+  };
+
+  handleValidation = (): void => {
+    this.handleTitleValidation();
+    this.handleDescriptionValidation();
+    this.handleAssigneeValidation();
+  };
+
+  handleTitleValidation = (): void => {
+    const { title } = this.state;
+
+    if (!title || title.length < 3 || title.length > 50) {
+      this.setState({
+        titleError: "Please enter a title between 3 and 50 characters!",
```

Comment:
This works great. I have an idea that will make it a bit cleaner.

```
  handleDescriptionValidation = (): void => {
    const { description } = this.state;
    const isValid = !description || description.length < 3 || description.length > 255;

    this.setState({
      descriptionError: isValid ? message : null
    });
  };
```

Reply:
done